### ISSUE TITLE
Feature idea: run angular-console into docker container or add prefix to each command

### ISSUE THREAD

**Issue by Maden-maxi** [nrwl/nx-console#571]
I use angular-console in VSCode.
I've create nx project via nodejs docker container because I don't want to install node directly to my machine for saving memory.
So the problem is that when I run angular-console command, it runs on my machine and runs with fail due to missing node on my machine. I've found a solution by simply copy generated command and paste it into terminal and run it into docker container or by adding prefix to command like so:

```
docker exec container_name yarn <angular-console command>
```
But this solution quite inconvenient.
So I suggest to create some setting which will be add a prefix to each command or some setting which in which we can log angular-console in into docker container to do angular-console able to run commands into docker container.

**Comment by mrmeku** [comment]
@Maden-maxi We are planning on adding NVM support in a future release. Would that fit your use case of not wanting to instal node on you system?

**Comment by Maden-maxi** [comment]
@mrmeku It will not fit my case, I meant to give ability to run angular-console commands in docker container environment , in this way will be using a correct version of node and also ability to use angular-console command without node.js on your machine by using node.js from docker node.js or any custom image. So in this case I won't worry about which version of node installed on my machine or which node version chosen by nvm on my machine.
To solve this problem I suggest add an option field in the settings.
I can't run 
```
ng <some command>
```
Till I install @angular/cli globally, so the ability to give a prefix or pre-command to each angular-console command gives ability to run angular-console command in any environment
So if for example I do not have installed globally @angular/cli I can do like so:
```
npm run ng <some command>
// or using yarn
yarn ng <some command>
``` 
Also in case when I don't have even node.js on my machine, I can use for example docker container
```
docker exec node_container (npm run or yarn) ng <some-command>
```

So basically it might work like this:

![settigns](https://user-images.githubusercontent.com/17612707/54475766-16206d80-47fe-11e9-953c-b5ab15ba4570.jpg)

![generate](https://user-images.githubusercontent.com/17612707/54475756-f5581800-47fd-11e9-86fe-480f27f8e600.jpg)

**Comment by mrmeku** [comment]
@Maden-maxi We won't be able to do this feature until we re-architect things a bit on windows. In Unix/Mac OS x we essentially run commands as bash scripts. But on windows we directly run the executable itself (which does not allow for prefixing)

**Comment by Cammisuli** [comment]
I thought about this for a while, and I think it’s currently outside the scope that we have planned for the extension now. 

If you are using docker containers for development, I recommend using the docker extension for VS Code and connecting to the container. You should then be able to use all the commands, and it’ll be executed in that container.  

More info here:
https://code.visualstudio.com/docs/remote/containers

As of now, I’m going to close this because there’s ways to do what’s being asked provided by another extension
